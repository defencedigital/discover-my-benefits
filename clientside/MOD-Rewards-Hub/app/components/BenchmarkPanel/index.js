import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'reactstrap';

import CopyCol from '../CopyCol';
import ContainerInner from '../ContainerInner';

import { formatCurrency } from '../../utils/currency';

const BenchmarkPanel = props => (
  <div className="benchmark-panel">
    <ContainerInner className="no-padding-lrg">
      <Row>
        <Col xs="12"></Col>
      </Row>
      <Row>
        <Col md="12" lg="8">
          <h3 className="h2 benchmark-panel-title">{props.title}</h3>
          {props.content && <CopyCol content={props.content} />}
        </Col>
        <Col md="12" lg="4">
          <span className="benchmark-panel-figure">{formatCurrency(props.benchmark)}</span>
        </Col>
      </Row>
    </ContainerInner>
  </div>
);

BenchmarkPanel.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
  benchmark: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

export default BenchmarkPanel;
