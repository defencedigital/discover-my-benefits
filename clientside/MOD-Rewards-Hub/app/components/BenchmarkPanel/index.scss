.benchmark-panel{
  margin: 40px 0 0;
  position: relative;

  .container-inner {
    padding: 40px 0;
    border-top: 1px solid $gray;
  }

  @include media-breakpoint-down(md) {

    padding: 40px 0 0;
    margin: 40px 0 0;

    .copy-col{
      padding-bottom: 0;
    }
  }
}

.benchmark-panel-title{
  margin-bottom: 40px;
}

.benchmark-panel-figure{
  padding: 10px 20px;
  display: block;
  width: 100%;
  padding-top: 30px;
  font-size: 4rem;
  font-weight: bold;
  text-align: center;
  margin: 0 auto;

  @include media-breakpoint-down(md) {
    text-align: left;
    padding: 10px 0;
  }
}

