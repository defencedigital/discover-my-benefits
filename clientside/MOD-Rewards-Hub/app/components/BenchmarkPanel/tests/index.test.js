import React from 'react';
import { fromJS } from 'immutable';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import BenchmarkPanel from '../index';

Enzyme.configure({ adapter: new Adapter() });

const mockedProps = fromJS({
  title: 'title',
  content: 'content',
  benchmark: 'benchmark',
});

describe('BenchmarkPanel component', () => {
  it('matches the snapshot', () => {
    const html = shallow(<BenchmarkPanel {...mockedProps.toJS()} />);
    expect(html).toMatchSnapshot();
  });
});
