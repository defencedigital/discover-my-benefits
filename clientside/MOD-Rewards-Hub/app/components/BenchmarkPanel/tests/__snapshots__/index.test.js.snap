// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BenchmarkPanel component matches the snapshot 1`] = `ShallowWrapper {}`;
